dask_geopandas/sjoin.py

```diff
@@ -0,0 +1,62 @@
+import numpy as np
+import geopandas
+
+from dask.base import tokenize
+from dask.highlevelgraph import HighLevelGraph
+
+from .core import from_geopandas, GeoDataFrame
+
+
+def sjoin(left, right, how="inner", op="intersects"):
+    """
+    Spatial join of two GeoDataFrames.
+
+    ...
+    """
+    if isinstance(left, geopandas.GeoDataFrame):
+        left = from_geopandas(left, npartitions=1)
+    if isinstance(right, geopandas.GeoDataFrame):
+        right = from_geopandas(right, npartitions=1)
+
+    name = "sjoin-" + tokenize(left, right, how, op)
+    meta = geopandas.sjoin(left._meta, right._meta, how=how, op=op)
+
+    if left.spatial_partitions is not None and right.spatial_partitions is not None:
+        # Spatial partitions are known -> use them to trim down the list of
+        # partitions that need to be joined
+        parts = geopandas.sjoin(
+            left.spatial_partitions.to_frame("geometry"),
+            right.spatial_partitions.to_frame("geometry"),
+            how="inner",
+            op="intersects",
+        )
+        parts_left = np.asarray(parts.index)
+        parts_right = np.asarray(parts["index_right"].values)
+        using_spatial_partitions = True
+    else:
+        # Unknown spatial partitions -> full cartesian (cross) product of all
+        # combinations of the partitions of the left and right dataframe
+        n_left = left.npartitions
+        n_right = right.npartitions
+        parts_left = np.repeat(np.arange(n_left), n_right)
+        parts_right = np.tile(np.arange(n_right), n_left)
+        using_spatial_partitions = False
+
+    dsk = {}
+    new_spatial_partitions = []
+    for i, (l, r) in enumerate(zip(parts_left, parts_right)):
+        dsk[(name, i)] = (geopandas.sjoin, (left._name, l), (right._name, r), how, op)
+        # TODO preserve spatial partitions of the output if only left has spatial
+        # partitions
+        if using_spatial_partitions:
+            lr = left.spatial_partitions.iloc[l]
+            rr = right.spatial_partitions.iloc[r]
+            # extent = lr.intersection(rr).buffer(buffer).intersection(lr.union(rr))
+            extent = lr.intersection(rr)
```

Comment:
@mrocklin I know it's a very long time ago ;) But do you remember what the rationale was for intersecting with the union of both regions? (the `.intersection(lr.union(rr))`)